examples/ieciidc-talker.c

```diff
@@ -0,0 +1,266 @@
+	case 'i':
+		strncpy(ifname, arg, sizeof(ifname) - 1);
+		break;
+	case 'm':
+		max_transit_time = atoi(arg);
+		break;
+	case 'p':
+		priority = atoi(arg);
+		break;
+	}
+
+	return 0;
+}
+
+static struct argp argp = { options, parser };
+
+static void init_pdu(struct avtp_stream_pdu *pdu)
+{
+	int res;
+
+	res = avtp_ieciidc_pdu_init(pdu, AVTP_IECIIDC_TAG_CIP);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_TV, 0);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_STREAM_ID,
+								STREAM_ID);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu,
+			AVTP_IECIIDC_FIELD_STREAM_DATA_LEN, STREAM_DATA_LEN);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_GV, 0);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_GATEWAY_INFO,
+									0);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_CHANNEL, 31);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_CIP_QI_1, 0);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_CIP_SID, 63);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_CIP_DBS, 6);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_CIP_FN, 3);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_CIP_QPC, 0);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_CIP_SPH, 1);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_CIP_QI_2, 2);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_CIP_FMT, 32);
+	assert(res == 0);
+
+	res = avtp_ieciidc_pdu_set(pdu, AVTP_IECIIDC_FIELD_CIP_TSF, 0);
+	assert(res == 0);
+}
+
+int main(int argc, char *argv[])
+{
+	int fd, res;
+	struct sockaddr_ll sk_addr;
+	struct avtp_stream_pdu *pdu = alloca(PDU_SIZE);
+	struct avtp_ieciidc_cip_payload *pay =
+			(struct avtp_ieciidc_cip_payload *)pdu->avtp_payload;
+	uint8_t seq_num = 0;
+	uint8_t dbc = 0;
+
+	argp_parse(&argp, argc, argv, 0, NULL, NULL);
+
+	fd = create_talker_socket(priority);
+	if (fd < 0)
+		return 1;
+
+	res = setup_socket_address(fd, ifname, macaddr, ETH_P_TSN, &sk_addr);
+	if (res < 0)
+		goto err;
+
+	init_pdu(pdu);
+
+	while (1) {
+		ssize_t n;
+		uint32_t avtp_time;
+		struct avtp_ieciidc_cip_source_packet *sp;
+
+		memset(pay->cip_data_payload, 0, DATA_LEN);
+		sp = (struct avtp_ieciidc_cip_source_packet *)
+							pay->cip_data_payload;
+
+		n = read(STDIN_FILENO, sp->cip_with_sph_payload,
+							MPEG_TS_PACKET_LEN);
+		if (n == 0)
+			break;
+
+		if (n != MPEG_TS_PACKET_LEN) {
+			fprintf(stderr, "read %zd bytes, expected %d\n",
+							n, MPEG_TS_PACKET_LEN);
+		}
+
+		res = calculate_avtp_time(&avtp_time, max_transit_time);
+		if (res < 0) {
+			fprintf(stderr, "Failed to calculate avtp time\n");
+			goto err;
+		}
+
+		/* There are no helpers for payload fields, so one must remember
+		 * of byte ordering */
```

Comment:
Check the coding style for multi-line comments: https://www.kernel.org/doc/html/latest/process/coding-style.html#commenting